### PR TITLE
AA-128: Course Home Outline Tab UI/UX improvements

### PR DESCRIPTION
**Pull request by ciduarte**
<p>Desktop Outline Tab before (first) and after (second)</p>
AA-128_courseHome_outlineTab_desktop_before
AA-128_courseHome_outlineTab_desktop_after
<p>Mobile Outline Tab before and after</p>
<div>
AA-128_courseHome_outlineTab_mobile_before
AA-128_courseHome_outlineTab_mobile_after
</div>

Here's a list of stuff this ticket was addressing because there's a lot going on in this PR!

**Header**
* Fixed words being cut off in title
* Added help button

**Welcome Message**
* Added TransitionReplace to improve the Show More/Less experience
* Replaced existing dismiss button with an IconButton (done in Alert)
* Removed border
* General UI improvements

**Course Goals**
* Simplified edit course goal widget to be a dropdown
* Removed edit state logic

**Outline**
* "Expand/Collapse All" button
* Incorporated IconButton in each section
* Updated completion icon and assignment link styling

### CODE REVIEW

**Review by Ahernkb — APPROVED** [review]
Looks good! Did we remove the assignment icon? (pencil/pad)

**Comment by ciduarte** [comment]
> Looks good! Did we remove the assignment icon? (pencil/pad)

Yes, Eugene's latest design settled on using the gray checkmark for all incomplete assignments.

File: src/course-header/Header.jsx [hunk 1]
```diff
@@ -78,13 +78,13 @@ function Header({
...
     <header className="course-header">
-      <div className="container-fluid py-2 d-flex align-items-center ">
+      <div className="container-fluid py-2 d-flex align-items-center">
         {headerLogo}
         <div className="flex-grow-1 course-title-lockup" style={{ lineHeight: 1 }}>
           <span className="d-block small m-0">{courseOrg} {courseNumber}</span>
           <span className="d-block m-0 font-weight-bold course-title">{courseTitle}</span>
         </div>
-
+        <a className="text-gray-700 mr-3" href="https://support.edx.org">{intl.formatMessage(messages.help)}</a>
```
Comment (mikix) [inline src/course-header/Header.jsx]: That url should come from config, like our other URLs. I know we're slipping a little bit on the configurability of the social media stuff, but this feels more important?

File: src/course-home/outline-tab/OutlineTab.jsx [hunk 2]
```diff
@@ -117,10 +124,18 @@ function OutlineTab({ intl }) {
               ...offerAlert,
             }}
           />
+          <div className="row w-100 m-0 mb-3 justify-content-end">
+            <div className="col-12 col-sm-auto p-0">
+              <Button variant="outline-primary" block onClick={() => { setExpandAll(!expandAll); }}>
+                {expandAll ? 'Collapse All' : 'Expand All'}
```
Comment (mikix) [inline src/course-home/outline-tab/OutlineTab.jsx]: Should be translated.

File: src/course-home/outline-tab/SequenceLink.jsx [hunk 3]
```diff
@@ -39,26 +39,43 @@ export default function SequenceLink({
 
   let text = title;
 
-  let faIcon;
-  switch (icon) {
```
Comment (mikix) [inline src/course-home/outline-tab/SequenceLink.jsx:43]: Yay! :fire:

File: src/course-home/outline-tab/messages.js [hunk 4]
```diff
@@ -5,6 +5,26 @@ const messages = defineMessages({
...
+  closeSection: {
+    id: 'learning.outline.altText.closeSection',
+    defaultMessage: 'Close',
+    description: 'A button to close the given section of the course outline',
+  },
+  closeToast: {
+    id: 'learning.outline.altText.closeToast',
+    defaultMessage: 'Close',
+    description: 'A button to close the given Toast',
+  },
```
Comment (mikix) [inline src/course-home/outline-tab/messages.js]: I dunno if these need two different strings. They would never be translated differently (I'm going to bet money no language changes the spelling of "close" based on whether it's in an outline or a toast context.)

Doesn't hurt anything, really. Just saying.

File: src/course-home/outline-tab/messages.js [hunk 5]
```diff
@@ -5,6 +5,26 @@ const messages = defineMessages({
...
+  completedAssignment: {
+    id: 'learning.outline.completedAssignment',
+    defaultMessage: 'Completed',
+    description: 'Text used to describe the green checkmark icon in front of an assignment title',
+  },
+  completedSection: {
+    id: 'learning.outline.completedSection',
+    defaultMessage: 'Completed section',
```
Comment (mikix) [inline src/course-home/outline-tab/messages.js:20]: Oooh, so "section" is a word we expose to learners? Is that new?

File: src/course-home/outline-tab/widgets/WelcomeMessage.jsx [hunk 6]
```diff
@@ -34,27 +35,36 @@ function WelcomeMessage({ courseId, intl }) {
...
                   : intl.formatMessage(messages.welcomeMessageShowLessButton)}
-              </button>
+              </Button>
             </div>
-          )
-        }
+          </div>
+        )}
+      >
+        <TransitionReplace className="mb-3" enterDuration={200} exitDuration={200}>
```
Comment (mikix) [inline src/course-home/outline-tab/widgets/WelcomeMessage.jsx:53]: Ooh fun, didn't know this one.

File: src/generic/user-messages/messages.js [hunk 7]
```diff
@@ -0,0 +1,11 @@
+import { defineMessages } from '@edx/frontend-platform/i18n';
+
+const messages = defineMessages({
+  closeAlert: {
+    id: 'general.alert.altText.closeAlert',
+    defaultMessage: 'Close',
```
Comment (mikix) [inline src/generic/user-messages/messages.js]: Here again -- optional, but if we could condense them somewhere, would be nice.

File: src/tab-page/messages.js [hunk 8]
```diff
@@ -1,6 +1,11 @@
 import { defineMessages } from '@edx/frontend-platform/i18n';
 
 const messages = defineMessages({
+  closeToast: {
+    id: 'learning.loading.altText.closeToast',
+    defaultMessage: 'Close',
```
Comment (mikix) [inline src/tab-page/messages.js]: I'm dying here.

**Review by mikix — COMMENTED** [review]
Got some just quick code review comments. I'll also run it locally and play around.

File: src/generic/user-messages/messages.js (hunk 7 above)
Comment (ciduarte) [reply src/generic/user-messages/messages.js]: Yeah I had the same thought.. I just wasn't sure where to put them if we were to condense them 😕

File: src/course-home/outline-tab/messages.js (hunk 4 above)
Comment (ciduarte) [reply src/course-home/outline-tab/messages.js]: Ooh good callout, the `closeSection` message used to be "Close section" and then I changed it w/o consolidating. I agree with you about the context

**Review by mikix — APPROVED** [review]

**Pull request merged**